gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -1,52 +1,190 @@
+                          .getOptionalPositiveInteger(PARAM_PORT_PRIMARY)
+                          .or(DEFAULT_PORT_PRI),
+                      "http"),
+                  new HttpHost(
+                      parameters.getOptionalString(PARAM_HOSTNAME_SECONDARY).or(DEFAULT_HOST),
+                      parameters
+                          .getOptionalPositiveInteger(PARAM_PORT_SECONDARY)
+                          .or(DEFAULT_PORT_SEC),
+                      "http")));
+
+      String indexName = parameters.getString(PARAM_INDEX_NAME);
+
+      if (parameters.isPresent(PARAM_GIGAWORD_DIRECTORY_PATH)) {
+        File gigawordDir = parameters.getExistingDirectory(PARAM_GIGAWORD_DIRECTORY_PATH);
+        for (File concatenatedFileToIndex : gigawordDir.listFiles()) {
+          if (concatenatedFileToIndex.isFile() && isValidGzipFile(concatenatedFileToIndex)) {
+            index(client, concatenatedFileToIndex, indexName);
+          }
         }
+      } else {
+        File concatenatedFileToIndex = parameters.getExistingFile(PARAM_GIGAWORD_FILEPATH);
+        if (!isValidGzipFile(concatenatedFileToIndex)) {
+        	client.close();
+        	throw new Exception(concatenatedFileToIndex.getAbsolutePath() + " is not a valid GZip file");
+        }
+        index(client, concatenatedFileToIndex, indexName);
+      }
+
+      client.close();
+
+    } catch (Exception e) {
+      log.error("Caught exception: {}", e);
+      System.exit(1);
+    }
+  }
+
+  public static boolean isValidGzipFile(File file) throws IOException {
+    // check file is a gzip file
+    String contentType = Files.probeContentType(file.toPath());
+    if ((contentType != null && !contentType.equalsIgnoreCase("application/gzip"))
+        || (contentType == null
+            && !FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("gz"))) {
+      return false;
     }
+    return true;
+  }
+
+  public static void index(RestHighLevelClient client, File file, String indexName)
+      throws Exception {
+
+    GigawordFileProcessor proc = new GigawordFileProcessor(file);
+    Iterator<Article> iterator = proc.iterator();
+
+    // Partition the iterator to size of 100
+    UnmodifiableIterator<List<Article>> partitions = Iterators.partition(iterator, 100);
+    while (partitions.hasNext()) {
+      List<Article> articles = partitions.next();
+      BulkRequest bulkRequest = new BulkRequest();
+      for (Article article : articles) {
+        XContentBuilder sourceBuilder =
+            buildSourceObject(article, "en", "", new Date().toString(), "");
+        bulkRequest.add(
+            new IndexRequest(indexName, "texts", article.getId()).source(sourceBuilder));
+      }
+
+      BulkResponse bulkResponse = client.bulk(bulkRequest, RequestOptions.DEFAULT);
+      if (bulkResponse.hasFailures()) {
+        throw new Exception(bulkResponse.buildFailureMessage());
+      }
+    }
+  }
+
+  private static XContentBuilder buildSourceObject(
```

Comment:
Add a comment noting that this is the format expected for Inception's ElasticSearch-based external search, and point to the relevant part of the Inception code

Reply:
Added.